cmd/sniproxy/sniproxy.go

```diff
@@ -83,12 +138,35 @@ func main() {
 		go s.promoteHTTPS(ln)
 	}
 
+	if *debugPort != 0 {
+		mux := http.NewServeMux()
+		tsweb.Debugger(mux)
+		srv := &http.Server{
+			Handler: mux,
+		}
+		dln, err := s.ts.Listen("tcp", fmt.Sprintf(":%d", *debugPort))
+		if err != nil {
+			log.Fatal(err)
+		}
+		go func() {
```

Comment:
I don't think we need an explicit `http.Server` here:
```go
go http.Serve(dln, mux)
```

(we also don't really care about the exit status, since it's just serving debug endpoints and it doesn't control the process lifecycle)

Reply:
Done.

Left the exit status because this is intended to be run within a user tailnet, and it will be the user who is scraping the metrics for their own monitoring. If the metrics export fails I think we do want to fail the service.